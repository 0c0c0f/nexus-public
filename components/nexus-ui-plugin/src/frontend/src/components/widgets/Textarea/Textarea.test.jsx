/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import {shallow} from 'enzyme';
import React from 'react';

import Textarea from './Textarea';
import FieldErrorMessage from "../FieldErrorMessage/FieldErrorMessage";

describe('Textfield', () => {
  const getTextarea = (extraProps) => {
    let props = {
      name: 'test',
      value: 'val',
      onChange: () => {
      },
      ...extraProps
    };
    return shallow(<Textarea {...props} />);
  };
  
  it('renders correctly', () => {
    expect(shallow(<Textarea/>)).toMatchSnapshot();
  });

  it('hides the error message by default', () => {
    const wrapper = getTextarea();

    expect(wrapper.containsMatchingElement(<FieldErrorMessage/>)).toBe(false);
    expect(wrapper.find('textarea').hasClass('missing-required-value')).toBe(false);
  });

  it('hides the error message when the value is required and not empty', () => {
    const wrapper = getTextarea({
      isRequired: true
    });

    expect(wrapper.containsMatchingElement(<FieldErrorMessage/>)).toBe(false);
    expect(wrapper.find('textarea').hasClass('missing-required-value')).toBe(false);
  });

  it('shows the error message when the value is required but empty', () => {
    const wrapper = getTextarea({
      isRequired: true,
      value: ''
    });

    expect(wrapper.containsMatchingElement(<FieldErrorMessage/>)).toBe(true);
    expect(wrapper.find('textarea').hasClass('missing-required-value')).toBe(true);
  });
});
