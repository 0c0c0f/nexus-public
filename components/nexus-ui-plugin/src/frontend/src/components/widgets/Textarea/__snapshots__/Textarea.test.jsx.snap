// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield renders correctly 1`] = `
<Fragment>
  <textarea
    className="nxrm-textarea"
  />
</Fragment>
`;
